luigi/contrib/bigquery_avro.py

```diff
@@ -86,10 +86,64 @@ def get_fields_with_description(bq_fields, avro_fields):
             new_fields = []
             for field in bq_fields:
                 avro_field = avro_fields[field[u'name']]
+                field_type = type(avro_field.type)
+
+                #Primitive and top level complex type support
                 field[u'description'] = avro_field.doc
-                if field[u'type'] == u'RECORD' and hasattr(avro_field.type, 'fields_dict'):
-                    field[u'fields'] = \
-                        get_fields_with_description(field[u'fields'], avro_field.type.fields_dict)
+
+                #Enum Support
+                if field_type is avro.schema.EnumSchema:
+                    field[u'description'] = avro_field.type.doc
+
+                #Record Support
+                if field_type is avro.schema.RecordSchema:
+                    field[u'description'] = avro_field.type.doc
+                    field[u'fields'] = get_fields_with_description(field[u'fields'], avro_field.type.fields_dict)
+
+                #Array Support
+                if field_type is avro.schema.ArraySchema:
+                    field[u'description'] = avro_field.type.items.doc
+                    field[u'fields'] = get_fields_with_description(field[u'fields'], avro_field.type.items.fields_dict)
+
+                #Union Support
+                if type(avro_field.type) is avro.schema.UnionSchema:
+                    for schema in avro_field.type.schemas:
+                        if type(schema) is avro.schema.RecordSchema:
+                            field[u'description'] = schema.doc
+                            field[u'fields'] = get_fields_with_description(field[u'fields'], schema.fields_dict)
+
+                #Map Support
+                if field_type is avro.schema.MapSchema:
+                    #directly bypass the key and value attributes in BigQuery Schema
```

Comment:
Sorry, I don't understand what "bypass" means in a context like this. We _are_ doing things with `...value...` on the following lines, so what was bypassed? Is this trying to explain the `-1`?

Reply:
Effectively what this does is Big Query generates a "key" attribute that doesn't have any direct interaction possible with the avro schema.  In this use case the code "skips" the key and directly starts work with the value attribute.  Cleaned up the comments to try and express that better.